### ISSUE TITLE
Afterglow sends MIDI messages when web cue buttons are used?

### ISSUE THREAD

**Issue by dandaka** [brunchboy/afterglow#46]
Where can I read about? I see messages in my Virtual output and they are giving positive feedback.

**Comment by dandaka** [comment]
Right now this code forces Sparke effect into loop, which can be stopped only once.

https://github.com/dandaka/lights-experiment/blob/master/src/lights/my_show.clj#L267-L269

Maybe it is because of positive feedback, or maybe something else, I need to figure it out.

**Comment by brunchboy** [comment]
Yes, Afterglow is designed to light up the cue on all of your mapped controllers when it is running, regardless of which one you use to start it, and darken it on all of them when it stops, and flash it on all of them when it is in the process of ending. It assumes that it should send MIDI messages to the controllers to do so, and the controllers should be configured in a mode in which they are only illuminated by remote MIDI messages, rather than lighting up under local control.

If your controller cannot be configured that way, you can tell Afterglow not to send it feedback messages when you configure the cue mapping for it, as described in the [add-midi-control-to-cue-mapping](http://deepsymmetry.org/afterglow/doc/afterglow.effects.cues.html#var-add-midi-control-to-cue-mapping) documentation. Quoting from that:

> Afterglow will attempt to provide feedback about the progress of the cue by sending note on/off or control-change values to the same controller when the cue starts and ends. The note velocities or control values used can be changed by passing in different values with `:feedback-on` and `:feedback-off`, and this behavior can be suppressed entirely by passing `false` with `:feedback-on`.

So if your controller can’t participate like this (which is a shame, it is really cool to have them all coordinated), when you map the cue to it add the parameters `:feedback-on false` to the argument list.

**Comment by brunchboy** [comment]
I have added some more discussion of this and links to the API documentation at various places in the introductory documentation, so hopefully it will be more easily discovered now.

I have finished all the refactoring and improvement I wanted to do of the MIDI support functions. I believe what I just pushed to Clojars is a good release candidate for version 0.1.6, so I hope you will be able to comment on a few more of these issues while I sleep, so I can do the release tomorrow. :grin:

**Comment by dandaka** [comment]
Works fine for me, thank you!

**Comment by brunchboy** [comment]
Afterglow 0.1.6 has been released. So much good stuff in it, and your ideas and requests were the reason for many of the fixes and new features, thank you!

In order to keep up with new development from me, you will want to update your `project.clj` so that you depend on `[org.clojure/clojure "1.8.0-RC5"]`, the latest (and probably final) Release Candidate of Clojure 1.8, which just came out today, and `[afterglow "0.2.0-SNAPSHOT"]`, which I have just created, and which will be where development of the next Afterglow release takes place.

Also, if you were using any of the functions which were deprecated in the last few releases of afterglow, you will need to update to their new locations and calling formats, as detailed in the [Change Log](https://github.com/brunchboy/afterglow/blob/master/CHANGELOG.md#unreleased), because I have removed the deprecated versions as part of starting development of version 0.2.0. Attempts to call the removed functions will fail to compile.

We are ready for the next iteration of new cool stuff!
